### PR TITLE
Issue 13 with avatars

### PR DESCRIPTION
**Pull request by polx**
This should solve issue #13 .

### CODE REVIEW

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml [hunk 1]
```diff
@@ -367,6 +367,36 @@
...
+              this.checked = false;
+            }
+            });
+            updateDriveNowWhat();
+          }
+
+          function updateDriveNowWhat() {
+            var driveInput = jQuery("input[name='scope_drive']")[0];
+            if(driveInput.checked) jQuery("#driveOnNowWhat").show();
+              else jQuery("#driveOnNowWhat").hide();
```
Comment (acotiuga) [inline src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: Even if there's only one instruction, it's recommended to use `{}`.

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml [hunk 2]
```diff
@@ -367,6 +367,36 @@
...
+            });
+            jQuery("input[name$='_scope']").val(scopes);
+            updateDriveNowWhat();
+          }
+
+          function readScopes() {
+            var scope = jQuery("input[name$='_scope']").val();
+            jQuery(("input[name^='scope_']")).each(function() {
+            if(jQuery.attr(this, "disabled")!=="disabled")
+            var n = this.name.substring('scope_'.length);
```
Comment (acotiuga) [inline src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: This line is not indented nor wrapped in `{}` so it's harder to read.

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml [hunk 3]
```diff
@@ -367,6 +367,36 @@
...
+              scopes = scopes + " " + this.name.substring('scope_'.length);
+            });
+            jQuery("input[name$='_scope']").val(scopes);
+            updateDriveNowWhat();
+          }
+
+          function readScopes() {
+            var scope = jQuery("input[name$='_scope']").val();
+            jQuery(("input[name^='scope_']")).each(function() {
+            if(jQuery.attr(this, "disabled")!=="disabled")
```
Comment (acotiuga) [inline src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: Bad indentation.

File: src/main/resources/GoogleApps/Translations.de.xml [hunk 4]
```diff
@@ -66,8 +66,8 @@ GoogleApps.GoogleAppsConfigClass_scope=Autorisierungen
 GoogleApps.GoogleAppsConfigClass_scope.hint=Welche Information sollte dieses Wiki Google abfragen?
 GoogleApps.GoogleAppsConfigClass_scope_identity=Vor- und Nachname.
 GoogleApps.GoogleAppsConfigClass_scope_email=Emailadresse.
-GoogleApps.GoogleAppsConfigClass_scope_gdoc=Google Docs lesen und speichern.
-GoogleApps.GoogleAppsConfigClass_scope_readAvatar=Benutzerprofilbild
```
Comment (acotiuga) [inline src/main/resources/GoogleApps/Translations.de.xml]: It's strongly recommended to not delete translation keys. They should be moved at the bottom of the page, in a `deprecated` area. https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HTranslationPropertyDeprecations.

File: src/main/resources/GoogleApps/Groovy.xml [hunk 5]
```diff
@@ -410,7 +410,7 @@ public class GoogleAppsGroovy {
             try {
                 // Force main wiki database to create the user as global
                 context.setDatabase("xwiki")
-                email = (user.emails!=null &amp;&amp; user.emails.size()&gt;0) ? user.emails.getAt(0).value : "";
+                email = (user.emails!=null &amp;&amp; user.emails.size()>0) ? user.emails.getAt(0).value : "";
```
Comment (acotiuga) [inline src/main/resources/GoogleApps/Groovy.xml]: Is this change the result of xar page export? As `&` was resolved to `&amp;`, so the `>` should be resolved as `&gt;`.

File: src/main/resources/GoogleApps/DriveMacro.xml [hunk 6]
```diff
@@ -295,97 +295,101 @@ def force = false;
 def gagroovy = xwiki.parseGroovyFromPage("GoogleApps.Groovy")
 gagroovy.init(xwiki, xcontext, doc)
 
-// adding stylesheet
-xwiki.ssx.use("GoogleApps.DriveMacro");
+if(!gagroovy.useDrive) {
+        println "This macro is disabled."
```
[the remark this answers is not in the document]
Comment (polx) [reply src/main/resources/GoogleApps/DriveMacro.xml]: fixed.

File: src/main/resources/GoogleApps/EditInGoogleApps.xml [hunk 7]
```diff
@@ -45,27 +45,29 @@
...
+  This document is not yet uploaded to Google Drive. Please confirm that you would like to upload it to edit it.
 
-(% class="buttonwrapper" %)[[$services.localization.render("googleapps.edit.retrieve.button")&gt;&gt;RetrieveFromGoogleApps||queryString="page=${escapetool.url($request.page)}&amp;name=${escapetool.url($request.name)}&amp;url=${escapetool.url($gdoc.exportLink)}&amp;id=${gdoc.id}"]]
+  (% class="buttonwrapper" %)[[$services.localization.render("googleapps.edit.confirm")&gt;&gt;||queryString="page=${escapetool.url($request.page)}&amp;name=${escapetool.url($request.name)}&amp;confirm=1"]]
+  #end
 #else
-This document is not yet uploaded to Google Drive. Please confirm that you would like to upload it to edit it.
-
-(% class="buttonwrapper" %)[[$services.localization.render("googleapps.edit.confirm")&gt;&gt;||queryString="page=${escapetool.url($request.page)}&amp;name=${escapetool.url($request.name)}&amp;confirm=1"]]
+    This macro is disabled.
```
[the remark this answers is not in the document]
Comment (polx) [reply src/main/resources/GoogleApps/EditInGoogleApps.xml]: fixed

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml (hunk 1 above)
Comment (polx) [reply src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: fixed.

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml (hunk 2 above)
Comment (polx) [reply src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: thanks for the catch. Fixed.

File: src/main/resources/GoogleApps/GoogleAppsConfigClass.xml (hunk 3 above)
Comment (polx) [reply src/main/resources/GoogleApps/GoogleAppsConfigClass.xml]: fixed.

File: src/main/resources/GoogleApps/Translations.de.xml (hunk 4 above)
Comment (polx) [reply src/main/resources/GoogleApps/Translations.de.xml]: fixed. thanks.

File: src/main/resources/GoogleApps/Groovy.xml (hunk 5 above)
Comment (polx) [reply src/main/resources/GoogleApps/Groovy.xml]: fixed.

File: src/main/resources/GoogleApps/EditInGoogleApps.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (polx) [reply src/main/resources/GoogleApps/EditInGoogleApps.xml]: Not yet done. There's a lot more macro text to be translated. Let's make this after we have a tested functionality.
paul

**Pull request merged**